common/types/timestamp.go

```diff
@@ -272,12 +274,33 @@ func timestampGetMillisecondsWithTz(t time.Time, tz ref.Val) ref.Val {
 
 func timeZone(tz ref.Val, visitor timestampVisitor) timestampVisitor {
 	return func(t time.Time) ref.Val {
+		var err error
+		var offset int
 		if StringType != tz.Type() {
 			return ValOrErr(tz, "no such overload")
 		}
-		loc, err := time.LoadLocation(string(tz.(String)))
-		if err == nil {
-			return visitor(t.In(loc))
+		val := string(tz.(String))
+		ind := strings.Index(val, ":")
+		if ind == -1 {
+			loc, err := time.LoadLocation(val)
+			if err == nil {
+				return visitor(t.In(loc))
+			}
+		} else {
+			hr, err := strconv.Atoi(string(val[0:ind]))
+			min, err2 := strconv.Atoi(string(val[ind+1]))
+			if err == nil && err2 == nil {
```

Comment:
Little bug here - if err == nil but err2 is non-nil, we'll end up returning an Err with an empty reason.

Instead, test err right after it's set and split out the error case, then instead of err2, just reuse err for the "min" parsing and also check for errors there.